app/src/http/handleError.ts

```diff
@@ -0,0 +1,23 @@
+import {useHistory} from 'react-router-dom';
+
+/**
+ * Custom hook to handle API errors and navigate to appropriate pages.
+ * @returns {function} handleError - Function to handle the error and perform navigation.
+ */
+const useHandleError = function (): { handleError: (errorCode: number) => void } {
+    const navigate = useHistory();
+
+    const handleError = (errorCode: number): void => {
+        switch (errorCode) {
+            case 404:
+                navigate.push('/404');
```

Comment:
It may be good to send some more error information to the page? Still looking to see how this is used but in the case I were searching for a recipe or equivalent you may not want to be thrown to a 404 page and would maybe rather have a 'no recipes found'.